enhancements/microshift/microshift-router-namespaces.md

```diff
@@ -0,0 +1,119 @@
+---
+title: microshift-route-admission-policy
+authors:
+  - "@pacevedom"
+reviewers:
+  - "@eslutsky"
+  - "@copejon"
+  - "@ggiguash"
+  - "@pmtk"
+  - "@pliurh"
+  - "@jerpeter1"
+  - "@Miciah"
+approvers:
+  - "@dhellmann"
+creation-date: 2024-01-30
+last-updated: 2024-01-31
+tracking-link:
+  - https://issues.redhat.com/browse/OCPSTRAT-1067
+see-also:
+  - "/enhancements/ingress/openshift-route-admission-policy.md"
+---
+
+# MicroShift route admission policy
+## Summary
+OpenShift defaults to not allow routes in multiple namespaces use the same
+hostname, and MicroShift inherits that default.
+
+Ever since OpenShift 4 this has been possible to configure, and MicroShift
+should allow that too to accommodate additional use cases.
+
+## Motivation
+Users who compose applications from multiple sources may deploy those to
+different namespaces, albeit want to expose them using the same hostname and
+different path.
+
+### User Stories
+As a MicroShift admin, I want to configure MicroShift to support combining
+routes from multiple namespaces.
+
+### Goals
+- Allow the use of the same host in routes from different namespaces.
+
+### Non-Goals
+N/A
+
+## Proposal
+The proposal follows all the changes that were done in the original
+[enhancement](https://github.com/openshift/enhancements/enhancements/ingress/openshift-route-admission-policy.md).
+
+This includes the exposure of a new configuration option to pass on to the
+router to allow cross-namespace host routes.
+
+### Workflow Description
+
+### API Extensions
+To allow configuring the router a new option is exposed through the MicroShift
+configuration file:
+```yaml
+router:
+  namespaceOwnership: <Strict|InterNamespaceAllowed> # Defaults to Strict.
```

Comment:
This is called RouteAdmission in the Ingress API. Why use a different name here?

Reply:
We may not want to import that API (I don't know what dependencies it would introduce). I just meant if we could create a struct that looks the same then we'd be consistent.

Reply:
Yes, thats what I meant, make it look the same without importing it.